File: algorithms/implementation/sherlock-and-squares/sherlock-and-squares-python/tests/test_squares.py


from squares import squares

class TesteSquares:
    
    def testeCaso1(self):
        
        a = 3
        b = 9
        
        esperado = 2
        
        resultado = squares(a, b)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        a = 17
        b = 24
        
        esperado = 0
        
        resultado = squares(a, b)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        a = 11
        b = 734
        
        esperado = 24
        
        resultado = squares(a, b)
        
        assert resultado == esperado
        
        
    def testeCaso4(self):
        
        a = 228
        b = 919
        
        esperado = 15
        
        resultado = squares(a, b)
        
        assert resultado == esperado

File: algorithms/implementation/sherlock-and-squares/sherlock-and-squares-javascript/js/test_squares.js
import {squares} from '../js/squares.js';
import assert from 'assert';

describe('squares', function() {

    it('O limite superior do intervalo é: 2', function() {

        let a = 3;
        let b = 9;

        let esperado = 2;

        let resultado = squares(a, b);

        assert.strictEqual(resultado, esperado);
    });
});

describe('squares', function() {

    it('O limite superior do intervalo é: 0', function() {

        let a = 17;
        let b = 24;

        let esperado = 0;

        let resultado = squares(a, b);

        assert.strictEqual(resultado, esperado);
    });
});

describe('squares', function() {

    it('O limite superior do intervalo é: 24', function() {

        let a = 11;
        let b = 734;

        let esperado = 24;

        let resultado = squares(a, b);

        assert.strictEqual(resultado, esperado);
    });
});

describe('squares', function() {

    it('O limite superior do intervalo é: 15', function() {

        let a = 228;
        let b = 919;

        let esperado = 15;

        let resultado = squares(a, b);

        assert.strictEqual(resultado, esperado);
    });
});